### PR TITLE
ilspycmd: init at 8.0

### PR DESCRIPTION
**Pull request by emilytrau**
###### Description of changes

Tool for decompiling .NET assemblies and generating portable PDBs https://github.com/icsharpcode/ILSpy

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/ilspycmd/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}:
+buildDotnetModule rec {
+  pname = "ilspycmd";
+  version = "8.0";
+
+  src = fetchFromGitHub {
+    owner = "icsharpcode";
+    repo = "ILSpy";
+    rev = "v${version}";
+    sha256 = "ERBYXgpBRXISfqBSBEydEQuD/5T1dvJ+wNg2U5pKip4=";
```
Comment (natsukium) [inline pkgs/development/tools/ilspycmd/default.nix:18]: Would you please use SRI hash?
> Nixpkgs contributors are currently recommended to use hash.

https://nixos.org/manual/nixpkgs/unstable/#fetchurl

File: pkgs/development/tools/ilspycmd/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  nativeBuildInputs = [
+    powershell
+  ] ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) [ autoSignDarwinBinariesHook ];
+
+  # https://github.com/NixOS/nixpkgs/issues/38991
+  # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
+  LOCALE_ARCHIVE = lib.optionalString stdenv.hostPlatform.isLinux "${glibcLocales}/lib/locale/locale-archive";
```
Comment (natsukium) [inline pkgs/development/tools/ilspycmd/default.nix:27]:
```suggestion
  env.LOCALE_ARCHIVE = lib.optionalString stdenv.hostPlatform.isLinux "${glibcLocales}/lib/locale/locale-archive";
```

**Review by natsukium — COMMENTED** [review]
I can build it fine with x86_64-{linux,darwin}.

File: pkgs/development/tools/ilspycmd/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
+  dotnet-runtime = dotnetCorePackages.runtime_6_0;
+
+  projectFile = "ICSharpCode.ILSpyCmd/ICSharpCode.ILSpyCmd.csproj";
+  nugetDeps = ./deps.nix;
+
+  meta = with lib; {
+    description = "Tool for decompiling .NET assemblies and generating portable PDBs";
+    homepage = "https://github.com/icsharpcode/ILSpy";
```
Comment (natsukium) [inline pkgs/development/tools/ilspycmd/default.nix:37]:
```suggestion
    homepage = "https://github.com/icsharpcode/ILSpy";
    changelog = "https://github.com/icsharpcode/ILSpy/releases/tag/${src.rev}";
```

I would be happy to add the `meta.changelog`.

Comment (emilytrau) [reply pkgs/development/tools/ilspycmd/default.nix:37]: Thanks for the review!

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 244761` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ilspycmd</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]

File: pkgs/development/tools/ilspycmd/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    hash = "sha256-ERBYXgpBRXISfqBSBEydEQuD/5T1dvJ+wNg2U5pKip4=";
+  };
+
+  nativeBuildInputs = [
+    powershell
+  ] ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) [ autoSignDarwinBinariesHook ];
+
+  # https://github.com/NixOS/nixpkgs/issues/38991
+  # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
+  env.LOCALE_ARCHIVE = lib.optionalString stdenv.hostPlatform.isLinux "${glibcLocales}/lib/locale/locale-archive";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ilspycmd/default.nix:27]: Could also be just a warning

File: pkgs/development/tools/ilspycmd/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  src = fetchFromGitHub {
+    owner = "icsharpcode";
+    repo = "ILSpy";
+    rev = "v${version}";
+    hash = "sha256-ERBYXgpBRXISfqBSBEydEQuD/5T1dvJ+wNg2U5pKip4=";
+  };
+
+  nativeBuildInputs = [
+    powershell
+  ] ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) [ autoSignDarwinBinariesHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ilspycmd/default.nix:23]:
```suggestion
  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [ autoSignDarwinBinariesHook ];
```
same thing but shorter

**Pull request merged**